relayer/src/supervisor.rs

```diff
@@ -245,12 +341,66 @@ impl Worker {
 
     /// Run the worker event loop.
     fn run(self, object: Object) {
-        let result = match object {
+        let result = match object.clone() {
             Object::UnidirectionalChannelPath(path) => self.run_uni_chan_path(path),
+            Object::Client(client) => self.run_client(client),
         };
 
         if let Err(e) = result {
-            eprintln!("worker error: {}", e);
+            eprintln!("[{}] worker error: {}", object.short_name(), e);
+        }
+        info!("[{}] worker exits", object.short_name());
+    }
```

Comment:
We could use a `span` here instead, cf. https://github.com/informalsystems/ibc-rs/pull/895/files#diff-98735cb33117650c9897aa6139df32fd48b36b9a74cd8982d2f9494f2e574a94R248-R256

```